application/classes/Ushahidi/Repository/Config.php

```diff
@@ -48,10 +48,19 @@ public function update(Entity $entity)
 		$this->verifyGroup($group);
 
 		$config = \Kohana::$config->load($group);
-		$immutable = $entity->getImmutable();
 
-		foreach ($entity->getChanged() as $key => $val) {
-			if (! in_array($key, $immutable)) {
+		$immutable = $entity->getImmutable();
+		foreach ($entity->getChanged() as $key => $val) {	
+			if (! in_array($key, $immutable)) {				
+				
+				/* Below is to reset the twitter-since_id when the search-terms are updated. This should be revised when the data-source tech-debt is addressed*/
+
+				if($key === 'twitter' && $val['twitter_search_terms'] !== $config['twitter']['twitter_search_terms'])
```

Comment:
I think for now this is an OK place to put it, but we should think about adding a better abstraction - however - if datasources are in need of a rewrite, it's worth noting it for then